### ISSUE TITLE
MongoDB 3.0 requires SCRAM-SHA-1

### ISSUE THREAD

**Issue by bingzer** [gordonmleigh/promised-mongo#35]
Needs to work with other auth provider.
I think right now, we only support MongoCR type. 
In MongoDB 3.0, SCRAM-SHA-1 is mandatory: [MongoDB 3.0 Security Improvements](http://docs.mongodb.org/manual/release-notes/3.0/#security-improvements)
### Database.js

```
server.addAuthProvider('mongocr', new MongoCR());
          // authenticate on connect
          server.on('connect', function (server) {
            server.auth('mongocr', config.dbName, config.auth.user, config.auth.password,
              function (error, server) {
                if (error) {
                  reject(error);
                } else {
                  resolve(server);
                }
              });
          });
```

**Comment by nodkrot** [comment]
+1

**Comment by benmliang** [comment]
+1

**Comment by nodkrot** [comment]
How long can this take?

**Comment by RangerMauve** [comment]
@nodkrot I ended up just switching over to use the official driver right out of the box. The returns from various APIs are a little different, but it supports returning promises if no callback is passed in.

**Comment by nodkrot** [comment]
Are you talking about this http://mongodb.github.io/node-mongodb-native/ ?

This is also an option https://github.com/petkaantonov/bluebird/blob/master/API.md#promisification

**Comment by RangerMauve** [comment]
@nodkrot Yup! That's the one.

**Comment by kylebakerio** [comment]
This is a dealbreaking, urgent issue. can't use mongolab until this is fixed... Trying to deploy my app and use mongolab, realized I had to upgrade mongo on my machine to 3 to interface with their stuff. Get all of that sorted out, and spent hours figuring out why I couldn't get pmongo to authenticate.

**Comment by irokhes** [comment]
+1

**Comment by kylebakerio** [comment]
Followed Ranger Mauve's advice and switched. Got to leave _most_ of my code intact. At least one thing did return something slightly different (an array instead of the value itself, iirc).

Had to use this code for the connection:

```
var MongoClient = require('mongodb').MongoClient
var assert = require('assert');
var collection;

// Standard Connection URL
var url = "mongodb://<username>:<password>@<databaseURL>";
// Use connect method to connect to the Server
MongoClient.connect(url, function(err, db) {
  assert.equal(null, err);
  console.log("Connected correctly to mongodb server");

  collection = db.collection('userData');

  // db.close();
});
```

Notice that collection is outside. That makes it accessible via closure to the functions in my exports object, leaving their code mostly intact.

**Comment by RangerMauve** [comment]
Yeah, from what I recall, `update`, `insert`, and `save` (probably a bunch of others that I'm just not using at the moment) have different returns, so you should take care to take that into account if you do the switch.

**Comment by haganbt** [comment]
+1. Lost so much time here trying to connect to MongoLab.

**Comment by azappa** [comment]
+1, me too, cannot connect to Mongolab when using mongodb v3 databases.

**Comment by haganbt** [comment]
@azappa - I ended up using the official driver. Here is an example that may help - https://github.com/haganbt/super-public-collector/blob/master/lib/db.js

**Comment by ocasta** [comment]
+1
